### PR TITLE
Geojson renderer

### PR DESCRIPTION
**Pull request by dgaubert**
cc/ @rochoa

### CODE REVIEW

File: lib/windshaft/renderers/mapnik/carto_psql_factory.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+var CartoPSQL = require('cartodb-psql');
+
+module.exports = function cartoPSQLFactory(dbParams, dbPoolParams) {
```
Comment (rochoa) [inline lib/windshaft/renderers/mapnik/carto_psql_factory.js]: I guess we created this for testing purposes. Is not possible to just mock cartodb-psql prototype?

File: lib/windshaft/renderers/mapnik/factory.js [hunk 2]
```diff
@@ -111,13 +116,63 @@ MapnikFactory.prototype.supportsFormat = function(format) {
...
+    var cartoPSQL = cartoPSQLFactory(dbParams, dbPoolParams);
+
+    var layerFilter = options.layer;
+    var mapLayers = mapConfig.getLayers();
+    var layers = [];
+
+    if (typeof layerFilter === 'number') {
+        layers = [ mapConfig.getLayer(layerFilter) ];
+    } else if (layerFilter && layerFilter !== 'mapnik') {
+        var filteredLayers = layerFilter.split(',').map(function(layerIdx) {
```
Comment (rochoa) [inline lib/windshaft/renderers/mapnik/factory.js]: All this code sounds to me like BlendRenderer. Could we extract the layer filtering/ordering?

File: lib/windshaft/renderers/mapnik/geojson_sql_wrapper.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+var fs = require('fs');
+var _ = require('underscore');
+var geojsonSQL = fs.readFileSync(__dirname + '/geojson_template.sql', 'utf8');
+
+function GeojsonSqlWrapper() {
+    this.geojsonTemplate = _.template(geojsonSQL);
```
Comment (rochoa) [inline lib/windshaft/renderers/mapnik/geojson_sql_wrapper.js]: We should use `dot` for anything template related.

**Comment by rochoa** [comment]
Let's ship it. Although I'm missing the buffer-size option.

**Comment by rochoa** [comment]
For the future we should consider using streaming responses.

File: lib/windshaft/renderers/mapnik/carto_psql_factory.js (hunk 1 above)
Comment (dgaubert) [inline lib/windshaft/renderers/mapnik/carto_psql_factory.js]: Absolutely!

File: lib/windshaft/renderers/mapnik/factory.js (hunk 2 above)
Comment (dgaubert) [inline lib/windshaft/renderers/mapnik/factory.js]: Ok!

File: lib/windshaft/renderers/mapnik/geojson_sql_wrapper.js (hunk 3 above)
Comment (dgaubert) [inline lib/windshaft/renderers/mapnik/geojson_sql_wrapper.js]: ok!

**Comment by rochoa** [comment]
GO! :+1:

**Comment by fdansv** [comment]
This is really exciting :heart:

**Pull request merged**